Resources/Prototypes/NPCs/utility_queries.yml

```diff
@@ -17,6 +34,23 @@
     - !type:TargetAccessibleCon
       curve: !type:BoolCurve
 
+- type: utilityQuery
+  id: NearbyGuns
+  query:
+    - !type:ComponentQuery
+      components:
+        - type: Gun
+        - type: Item
+  considerations:
+    # TODO: Prefer highest DPC probably?
```

Comment:
Should probably find some way to do multivariable optimization so it'll also consider how much ammo is available.